src/components/Table/components/cell.tsx

```diff
@@ -34,6 +34,8 @@ export type GridCellContent = {
   revealOnRowHover?: true;
   /** Tooltip to add to a cell */
   tooltip?: ReactNode;
+  /** Allows cell to be editable when hovering, and also highlights the field on hover */
+  editableOnHover?: true;
```

Comment:
nit: add a comment about how if true, the select field background defaults to white as opposed to our standard gray blah blah need to style the select field, use with caution etc.